app/controllers/stations_controller.rb

```diff
@@ -0,0 +1,67 @@
+class StationsController < ApplicationController
+	before_action :set_station, only: [:show, :edit, :update, :destroy]
+
+	def index
+		if params[:search].present?
+		  @stations = Station.near(params[:search], 50, :order => :distance)
```

Comment:
50 is hardcode value.
Base on map scale, we can have appropriate value here.
We can improve later.